app/controllers/identify_controller.rb

```diff
@@ -123,6 +124,11 @@ def log_certificate(cert)
       attributes.delete(:issuer)
     end
 
+    # Log certificate if it fails either OpenSSL validation, but passes our current validation or vice versa
+    if valid != login_certs_openssl_result[:valid] || valid != ficam_certs_openssl_result[:valid]
+      CertificateLoggerService.log_certificate(cert)
```

Comment:
**question:** does this provide any additional context as to why the cert failed in either direction? Or is the thought that we can just run it through once it's been logged and see for ourselves. If that's the case, it would be great to have a rake task to run a given cert through both checks and compare (if such a thing can be easily done locally)

Reply:
Yep, we log the status and error result for all certs: https://github.com/18F/identity-pki/blob/2ab09f5/app/controllers/identify_controller.rb#L113-L119

This change will help us see some of the details of the cert and be able to re-run verification, etc.

Reply:
An example log:

```json
{
  "name": "Certificate Processed",
  "signing_key_id": "AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA",
  "key_id": "AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA",
  "issuer": "ABC ISSUER",
  "card_type": "piv",
  "valid_policies": true,
  "valid": true,
  "error": null,
  "openssl_valid": false,
  "openssl_errors": "error 43 at 0 depth lookup: no explicit policy",
  "ficam_openssl_valid": false,
  "ficam_openssl_errors": "error 43 at 0 depth lookup: no explicit policy"
}
```